lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -187,17 +202,32 @@ def _loop_process_output(self, output):
             raise IOError("Couldn't check file size; %s" % err)
 
         time_passed = self.LOOP_TIMEOUT_INIT + self._loop_count * self.LOOP_TIMEOUT_MAIN
-        if time_passed > self.output_timeout and not self.seen_output:
-            warning_msg = '\n'.join([
-            "WARNING: mympirun has been running for %s seconds without seeing any output." % time_passed,
-            "This may mean that your program is hanging, please check and make sure that is not the case!",
-            '',
-            "If this warning is printed too soon and the program is doing useful work without producing any output,",
-            "you can increase the timeout threshold via --output-check-timeout (current setting: %s seconds)" % self.output_timeout,
-            ])
-            self.log.error(warning_msg + '\n')
+        self.seen_output = check_output(time_passed, self.output_timeout, self.seen_output)
+
+
+class RunAsyncMPI(RunAsyncLoopStdout):
+    """
+    Stream output to stdout as in RunAsyncLoopStdout
+    while also checking whether any output has been produced after a specified amount of time.
+
+    """
+    def __init__(self, cmd, **kwargs):
+        self.output_timeout = kwargs.pop('output_timeout', None)
+
+        super(RunAsyncMPI, self).__init__(cmd, **kwargs)
+
+        self.seen_output = self.output_timeout < 0 #no check when output_timeout is negative
+
+    def _loop_process_output(self, output):
+        """ Send output to stdout + hang check """
+        if len(output) > 0:
             self.seen_output = True
 
+        time_passed = self.LOOP_TIMEOUT_INIT + self._loop_count * self.LOOP_TIMEOUT_MAIN
+        self.seen_output = check_output(time_passed, self.output_timeout, self.seen_output)
```

Comment:
@Caylo passing `self.seen_output` and then updating it is silly... the `seen_output` logic should be handled here, not in the `check_output` function